python_modules/dagster/dagster/core/events/log.py

```diff
@@ -124,12 +118,23 @@ def from_json(json_str):
     def dagster_event_type(self):
         return self.dagster_event.event_type if self.dagster_event else None
 
+    def get_message(self) -> str:
+        """
+        Return the message from the structured DagsterEvent if present, fallback to user_message
+        """
+
+        if self.is_dagster_event:
+            msg = self.get_dagster_event().message
+            if msg is not None:
+                return msg
+
+        return self.user_message
```

Comment:
replicated here

Reply:
prefer the signature change rather than adding a property?

Reply:
not a bad idea - would reduces changes required